### ISSUE TITLE
Trigger build button displays slower as PR's increase

### ISSUE THREAD

**Issue by pemercer** [tomasbjerre/pull-request-notifier-for-bitbucket#250]
Bitbucket 4.14.5
PR Notifier 2.67
Bitbucket DC 5K users

 the button definition is retrieved from the REST endpoint
 
https://<server>/bitbucket/rest/prnfb-admin/1.0/settings/buttons/repository/{repo-num}/pullrequest/{pr-num}
 
some old PRs do show the button successfully (e.g., PRs 100 and 200). In these cases, the endpoint returns a response in 6-8 seconds in my trials:
 
https://<server>/bitbucket/rest/prnfb-admin/1.0/settings/buttons/repository/6940/pullrequest/100
https://<server>/bitbucket/rest/prnfb-admin/1.0/settings/buttons/repository/6940/pullrequest/200
 
Now if you try the same thing for one of the later PRs (I’m using PR 578), it takes 276-287 seconds in my trials to respond.

**Comment by tomasbjerre** [comment]
Perhaps same as https://github.com/tomasbjerre/pull-request-notifier-for-bitbucket/issues/237

**Comment by pemercer** [comment]
Can #237 be ported to BBS 4.x?

**Comment by tomasbjerre** [comment]
Yes it can. But Atlassian is making it hard to maintain several major versions so it is not a priority for me.

**Comment by pemercer** [comment]
Ok We will go ahead and move to BB 5.x this weekend and see if the new version helps.

**Comment by pemercer** [comment]
BB DC server is now upgraded to BB 5.3.2, other plugin that was filling the event queue has been removed.
PR Notifier version 3.15 is installed
Still seeing the slow response time for the configured button to show up.

**Comment by pemercer** [comment]
Is there a planned fix date, or additional information I can provide to help with debug?

**Comment by tomasbjerre** [comment]
No not right now.

**Comment by pemercer** [comment]
We will be moving to 5.6.1 in a few weeks, will there be a newer version of the plugin that may have this issue fixed by then?

**Comment by tomasbjerre** [comment]
I dont think so.

Den 14 dec. 2017 18:12 skrev "pemercer" <notifications@github.com>:
